python/mxnet/gluon/trainer.py

```diff
@@ -170,11 +178,63 @@ def step(self, batch_size, ignore_stale_grad=False):
         if not self._kv_initialized:
             self._init_kvstore()
 
+        self._allreduce(batch_size)
+        self._update(batch_size, ignore_stale_grad)
+
+    def allreduce(self, batch_size):
```

Comment:
I don't think batch_size is needed here. This api is only used when self._update_on_kvstore is false

Reply:
step and update should set rescale_grad, while allreduec, _allreduce, _update shouldn't

Reply:
Do you think allreduce or allreduce_grads is a better name here?

Reply:
allreduce_grads seems better.